### PR TITLE
chore: Re-enable all integration tests

### PR DESCRIPTION
**Pull request by jamtur01**
This PR re-enables the Humio and GCP tests and fixes some warnings and failures in the GCP tests.

@bruceg I am unsure how to address the humio tests as I am not quite sure I understand the relationship between the Spunk and Humio sinks.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -446,7 +446,8 @@ ifeq ($(AUTOSPAWN), true)
 	$(MAKE) start-integration-gcp
 	sleep 10 # Many services are very slow... Give them a sec..
 endif
-	${MAYBE_ENVIRONMENT_EXEC} cargo test --no-default-features --features gcp-integration-tests --lib ::gcp:: -- --nocapture
+	${MAYBE_ENVIRONMENT_EXEC} cargo test --no-default-features --features "gcp-integration-tests gcp-pubsub-integration-tests gcp-cloud-storage-integration-tests" \
```
Comment (bruceg) [inline Makefile:449]: I think `--features` can use comma to separate to avoid needing quoting.

**Review by bruceg — APPROVED** [review]

**Comment by jamtur01** [comment]
@fanatid The Humio failed tests:

```
error[E0599]: no method named `send` found for enum `sinks::VectorSink` in the current scope
   --> src/sinks/humio_logs.rs:161:14
    |
161 |         sink.send(event).compat().await.unwrap();
    |              ^^^^ method not found in `sinks::VectorSink`
    |
   ::: src/sinks/mod.rs:65:1
    |
65  | pub enum VectorSink {
    | ------------------- method `send` not found for this
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following traits define an item `send`, perhaps you need to implement one of them:
            candidate #1: `futures::sink::Sink`
            candidate #2: `futures_util::sink::SinkExt`
            candidate #3: `net2::ext::UdpSocketExt`

error[E0599]: no method named `send` found for enum `sinks::VectorSink` in the current scope
   --> src/sinks/humio_logs.rs:194:14
    |
194 |         sink.send(event).compat().await.unwrap();
    |              ^^^^ method not found in `sinks::VectorSink`
    |
   ::: src/sinks/mod.rs:65:1
    |
65  | pub enum VectorSink {
    | ------------------- method `send` not found for this
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following traits define an item `send`, perhaps you need to implement one of them:
            candidate #1: `futures::sink::Sink`
            candidate #2: `futures_util::sink::SinkExt`
            candidate #3: `net2::ext::UdpSocketExt`

error[E0599]: no method named `send` found for enum `sinks::VectorSink` in the current scope
   --> src/sinks/humio_logs.rs:225:18
    |
225 |             sink.send(event).compat().await.unwrap();
    |                  ^^^^ method not found in `sinks::VectorSink`
    |
   ::: src/sinks/mod.rs:65:1
    |
65  | pub enum VectorSink {
    | ------------------- method `send` not found for this
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following traits define an item `send`, perhaps you need to implement one of them:
            candidate #1: `futures::sink::Sink`
            candidate #2: `futures_util::sink::SinkExt`
            candidate #3: `net2::ext::UdpSocketExt`

error[E0599]: no method named `send` found for enum `sinks::VectorSink` in the current scope
   --> src/sinks/humio_logs.rs:246:18
    |
246 |             sink.send(event).compat().await.unwrap();
    |                  ^^^^ method not found in `sinks::VectorSink`
    |
   ::: src/sinks/mod.rs:65:1
    |
65  | pub enum VectorSink {
    | ------------------- method `send` not found for this
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following traits define an item `send`, perhaps you need to implement one of them:
            candidate #1: `futures::sink::Sink`
            candidate #2: `futures_util::sink::SinkExt`
            candidate #3: `net2::ext::UdpSocketExt`

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0599`.
error: could not compile `vector`.
```

Seem related to #3685.

**Review by jszwedko — COMMENTED** [review]
Looks good to me! Thanks for readding these.

I pushed a commit fixing the humio integration tests to be compatible with the changes in #2934

**Review by jszwedko — APPROVED** [review]

**Comment by jamtur01** [comment]
@jszwedko Thanks!

**Pull request merged**